spring-cloud-generator/showcase/showcase-spring-starter/src/test/java/com/google/showcase/v1beta1/spring/EchoAutoConfigurationTests.java

```diff
@@ -0,0 +1,369 @@
+        ctx -> {
+          EchoClient client = ctx.getBean(EchoClient.class);
+          assertThat(client).isNotNull();
+        });
+  }
+
+  @Test
+  void testCredentials_fromServicePropertiesIfSpecified() {
+    this.contextRunner
+        .withPropertyValues(
+            "com.google.showcase.v1beta1.echo.credentials.location=file:"
+                + SERVICE_CREDENTIAL_LOCATION)
+        .run(
+            ctx -> {
+              EchoClient client = ctx.getBean(EchoClient.class);
+              Credentials credentials =
+                  client.getSettings().getCredentialsProvider().getCredentials();
+              assertThat(((ServiceAccountCredentials) credentials).getClientId())
+                  .isEqualTo(SERVICE_CREDENTIAL_CLIENT_ID);
+            });
+  }
+
+  @Test
+  void testCredentials_fromTopLevelIfNoServiceProperties() {
+    this.contextRunner.run(
+        ctx -> {
+          EchoClient client = ctx.getBean(EchoClient.class);
+          Credentials credentials = client.getSettings().getCredentialsProvider().getCredentials();
+          assertThat(((ServiceAccountCredentials) credentials).getClientId())
+              .isEqualTo(TOP_LEVEL_CREDENTIAL_CLIENT_ID);
+        });
+  }
+
+  @Test
+  void testShouldGetTransportChannelProviderFromBeanWithQualifierName() throws IOException {
+    this.contextRunner
+        .withBean(
+            "anotherTransportChannelProvider",
+            TransportChannelProvider.class,
+            () -> mockTransportChannelProvider)
+        .run(
+            ctx -> {
+              assertThat(ctx.getBeanNamesForType(TransportChannelProvider.class))
+                  .containsExactlyInAnyOrder(
+                      "anotherTransportChannelProvider", TRANSPORT_CHANNEL_PROVIDER_QUALIFIER_NAME);
+              EchoClient client = ctx.getBean(EchoClient.class);
+              TransportChannelProvider transportChannelProviderBean =
+                  (TransportChannelProvider) ctx.getBean(TRANSPORT_CHANNEL_PROVIDER_QUALIFIER_NAME);
+              TransportChannelProvider transportChannelProvider =
+                  client.getSettings().getTransportChannelProvider();
+              assertThat(transportChannelProvider).isSameAs(transportChannelProviderBean);
+              assertThat(transportChannelProvider).isNotSameAs(mockTransportChannelProvider);
+            });
+  }
+
+  @Test
+  void testShouldUseDefaultTransportChannelProvider() {
+    this.contextRunner.run(
+        ctx -> {
+          EchoClient client = ctx.getBean(EchoClient.class);
+          TransportChannelProvider transportChannelProvider =
+              client.getSettings().getTransportChannelProvider();
+          TransportChannelProvider defaultTransportChannelprovider =
+              EchoSettings.defaultTransportChannelProvider();
+          assertThat(transportChannelProvider)
+              .usingRecursiveComparison()
+              .isEqualTo(defaultTransportChannelprovider);
+        });
+  }
+
+  //  @Test
```

Comment:
```
Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [com.google.showcase.v1beta1.EchoClient]: Factory method 'echoClient' threw exception with message: Cannot invoke "com.google.api.gax.rpc.ApiCallContext.withStreamWaitTimeout(org.threeten.bp.Duration)" because the return value of "com.google.api.gax.rpc.ApiCallContext.withStreamIdleTimeout(org.threeten.bp.Duration)" is null
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:171)
	at org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:655)
	... 99 more
Caused by: java.lang.NullPointerException: Cannot invoke "com.google.api.gax.rpc.ApiCallContext.withStreamWaitTimeout(org.threeten.bp.Duration)" because the return value of "com.google.api.gax.rpc.ApiCallContext.withStreamIdleTimeout(org.threeten.bp.Duration)" is null
	at com.google.api.gax.rpc.Callables.watched(Callables.java:119)
	at com.google.api.gax.grpc.GrpcCallableFactory.createServerStreamingCallable(GrpcCallableFactory.java:283)
	at com.google.showcase.v1beta1.stub.GrpcEchoCallableFactory.createServerStreamingCallable(GrpcEchoCallableFactory.java:102)
	at com.google.showcase.v1beta1.stub.GrpcEchoStub.<init>(GrpcEchoStub.java:317)
	at com.google.showcase.v1beta1.stub.GrpcEchoStub.<init>(GrpcEchoStub.java:226)
	at com.google.showcase.v1beta1.stub.GrpcEchoStub.create(GrpcEchoStub.java:208)
	at com.google.showcase.v1beta1.stub.EchoStubSettings.createStub(EchoStubSettings.java:388)
	at com.google.showcase.v1beta1.EchoClient.<init>(EchoClient.java:180)
	at com.google.showcase.v1beta1.EchoClient.create(EchoClient.java:163)
	at com.google.showcase.v1beta1.spring.EchoSpringAutoConfiguration.echoClient(EchoSpringAutoConfiguration.java:291)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:139)
	... 100 more
```

Reply:
Resolved in https://github.com/GoogleCloudPlatform/spring-cloud-gcp/pull/1957/commits/0bdd26d59dd427b6f6fc5c635fab8203c0b1bca0.